test/integration/dryrun/dryrun_test.go

```diff
@@ -0,0 +1,677 @@
+	gvr("apiregistration.k8s.io", "v1beta1", "apiservices"): {
+		stub: `{"metadata": {"name": "as1.foo.com"}, "spec": {"group": "foo.com", "version": "as1", "groupPriorityMinimum":100, "versionPriority":10}}`,
+	},
+	// --
+
+	// k8s.io/kube-aggregator/pkg/apis/apiregistration/v1
+	// depends on aggregator using the same ungrouped RESTOptionsGetter as the kube apiserver, not SimpleRestOptionsFactory in aggregator.go
+	gvr("apiregistration.k8s.io", "v1", "apiservices"): {
+		stub: `{"metadata": {"name": "as2.foo.com"}, "spec": {"group": "foo.com", "version": "as2", "groupPriorityMinimum":100, "versionPriority":10}}`,
+	},
+	// --
+
+	// k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1
+	gvr("apiextensions.k8s.io", "v1beta1", "customresourcedefinitions"): {
+		stub:       `{"metadata": {"name": "openshiftwebconsoleconfigs.webconsole.operator.openshift.io"},"spec": {"scope": "Cluster","group": "webconsole.operator.openshift.io","version": "v1alpha1","names": {"kind": "OpenShiftWebConsoleConfig","plural": "openshiftwebconsoleconfigs","singular": "openshiftwebconsoleconfig"}}}`,
+		skipUpdate: true,
+	},
+	// --
+
+}
+
+// Only add kinds to this list when this a virtual resource with get and create verbs that doesn't actually
+// store into it's kind.  We've used this downstream for mappings before.
+var kindWhiteList = sets.NewString()
+
+// namespace used for all tests, do not change this
+const testNamespace = "dryrunnamespace"
+
+func DryRunCreateTest(t *testing.T, rsc dynamic.ResourceInterface, obj *unstructured.Unstructured, gvResource schema.GroupVersionResource) {
+	if _, err := rsc.Create(obj, metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}}); err != nil {
+		t.Fatalf("failed to dry-run create stub for %s: %#v", gvResource, err)
+	}
+
+	if _, err := rsc.Get(obj.GetName(), metav1.GetOptions{}); err == nil {
+		t.Fatalf("object shouldn't exist")
+	}
+}
+
+func DryRunPatchTest(t *testing.T, rsc dynamic.ResourceInterface, name string) {
+	patch := []byte(`{"metadata":{"annotations":{"patch": "true"}}}`)
+	obj, err := rsc.Patch(name, types.MergePatchType, patch, metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}})
+	if err != nil {
+		t.Fatalf("failed to dry-run patch object: %v", err)
+	}
+	if _, ok := obj.GetAnnotations()["patch"]; !ok {
+		t.Fatalf("dry-run patched annotations should be returned, got: %v", obj.GetAnnotations())
+	}
+	obj, err = rsc.Get(obj.GetName(), metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get object: %v", err)
+	}
+	if _, ok := obj.GetAnnotations()["patch"]; ok {
+		t.Fatalf("dry-run patched annotations should not be persisted, got: %v", obj.GetAnnotations())
+	}
+}
+
+func DryRunStatusUpdateTest(t *testing.T, rsc dynamic.ResourceInterface, name string) {
+	obj, err := rsc.Get(name, metav1.GetOptions{}, "status")
+	if err == nil {
+		_, err = rsc.Update(obj, metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}})
```

Comment:
I think we still need to check the returned value got updated, and that the persisted object status didn't change

Reply:
I've replaced "Status" with "scale" so that I can do an actual change with a patch.